compiler/src/iree/compiler/Tools/iree_compile_lib.cc

```diff
@@ -161,28 +93,15 @@ int mlir::iree_compiler::runIreecMain(int argc, char **argv) {
       llvm::cl::desc("Verifies the IR for correctness throughout compilation."),
       llvm::cl::init(true));
 
-  llvm::cl::opt<IREEVMPipelinePhase> compileTo(
+  llvm::cl::opt<llvm::StringRef> compileTo(
       "compile-to",
       llvm::cl::desc(
           "Compilation phase to run up until before emitting output."),
-      llvm::cl::values(
-          clEnumValN(IREEVMPipelinePhase::Input, "input",
-                     "Performs input processing and lowering into core IREE "
-                     "input dialects (linalg/etc)."),
-          clEnumValN(
-              IREEVMPipelinePhase::ABI, "abi",
-              "Adjusts program ABI for the specified execution environment."),
-          clEnumValN(IREEVMPipelinePhase::Flow, "flow",
-                     "Compiles up to the `flow` dialect."),
-          clEnumValN(IREEVMPipelinePhase::Stream, "stream",
-                     "Compiles up to the `stream` dialect."),
-          clEnumValN(IREEVMPipelinePhase::HAL, "hal",
-                     "Compiles up to the `hal` dialect, including codegen."),
-          clEnumValN(IREEVMPipelinePhase::VM, "vm",
-                     "Compiles up to the `vm` dialect."),
-          clEnumValN(IREEVMPipelinePhase::End, "end",
-                     "Complete the full compilation pipeline.")),
-      llvm::cl::init(IREEVMPipelinePhase::End));
+      llvm::cl::init("end"));
+  enumerateIREEVMPipelinePhases(
+      [&](IREEVMPipelinePhase phase, StringRef name, StringRef desc) {
+        compileTo.getParser().addLiteralOption(name, name, desc);
+      });
```

Comment:
Something in here seems to have broken the `--compile-to=` flag:

```
D:\dev\projects\iree-build                                                                                          
λ .\tools\iree-compile.exe --compile-to=hal ..\iree\samples\models\simple_abs.mlir -o ../iree-tmp/simple_abs_hal.mli
r                                                                                                                   
no default value                                                                                                    
UNREACHABLE executed at D:\dev\projects\iree\third_party\llvm-project\llvm\include\llvm/Support/CommandLine.h:552!  
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace.         
Stack dump:                                                                                                         
0.      Program arguments: .\\tools\\iree-compile.exe --compile-to=hal ..\\iree\\samples\\models\\simple_abs.mlir -o
 ../iree-tmp/simple_abs_hal.mlir                                                                                    
Exception Code: 0x80000003                                                                                          
 #0 0x00007ffbd5298fc5 HandleAbort D:\dev\projects\iree\third_party\llvm-project\llvm\lib\Support\Windows\Signals.in
c:414:0                                                                                                             
 #1 0x00007ffc69921881 (C:\WINDOWS\System32\ucrtbase.dll+0x71881)                                                   
 #2 0x00007ffc69922851 (C:\WINDOWS\System32\ucrtbase.dll+0x72851)                                                   
 #3 0x00007ffbd5178ed7 llvm::llvm_unreachable_internal(char const *, char const *, unsigned int) D:\dev\projects\ire
e\third_party\llvm-project\llvm\lib\Support\ErrorHandling.cpp:212:0                                                 
 #4 0x00007ffbcc84372d llvm::cl::OptionValueBase<class llvm::StringRef, 1>::getValue(void) const D:\dev\projects\ire
e\third_party\llvm-project\llvm\include\llvm\Support\CommandLine.h:552:0                                            
 #5 0x00007ffbcc8440e8 llvm::cl::opt<class llvm::StringRef, 0, class llvm::cl::parser<class llvm::StringRef>>::handl
eOccurrence(unsigned int, class llvm::StringRef, class llvm::StringRef) D:\dev\projects\iree\third_party\llvm-projec
t\llvm\include\llvm\Support\CommandLine.h:1421:0                                                                    
 #6 0x00007ffbd5188ce0 llvm::cl::Option::addOccurrence(unsigned int, class llvm::StringRef, class llvm::StringRef, b
ool) D:\dev\projects\iree\third_party\llvm-project\llvm\lib\Support\CommandLine.cpp:1852:0                          
 #7 0x00007ffbd5182536 CommaSeparateAndAddOccurrence D:\dev\projects\iree\third_party\llvm-project\llvm\lib\Support\
CommandLine.cpp:647:0                                                                                               
 #8 0x00007ffbd5186fae ProvideOption D:\dev\projects\iree\third_party\llvm-project\llvm\lib\Support\CommandLine.cpp:
686:0                                                                                                               
 #9 0x00007ffbd5184c9b `anonymous namespace'::CommandLineParser::ParseCommandLineOptions D:\dev\projects\iree\third_
party\llvm-project\llvm\lib\Support\CommandLine.cpp:1711:0                                                          
#10 0x00007ffbd5186a5f llvm::cl::ParseCommandLineOptions(int, char const *const *, class llvm::StringRef, class llvm
::raw_ostream *, char const *, bool) D:\dev\projects\iree\third_party\llvm-project\llvm\lib\Support\CommandLine.cpp:
1470:0                                                                                                              
#11 0x00007ffbcc846c5f mlir::iree_compiler::runIreecMain(int, char **) D:\dev\projects\iree\compiler\src\iree\compil
er\Tools\iree_compile_lib.cc:131:0                                                                                  
#12 0x00007ff6b2ed15c8 __scrt_common_main_seh d:\a01\_work\43\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl
:288:0                                                                                                              
#13 0x00007ffc6b8374b4 (C:\WINDOWS\System32\KERNEL32.DLL+0x174b4)                                                   
#14 0x00007ffc6bc026a1 (C:\WINDOWS\SYSTEM32\ntdll.dll+0x526a1)                                                      
```

(maybe we should add a unit test for `--compile-to`)

Reply:
+1 to a test. I was relying on testing covering these things...